processor/probabilisticsamplerprocessor/sampler_mode.go

```diff
@@ -312,6 +504,24 @@ func commonShouldSampleLogic[T any](
 
 	sampled := threshold.ShouldSample(rnd.randomness())
 
+	if sampled && carrier != nil {
+		// Note: updateThreshold limits loss of adjusted count, by
+		// preventing the threshold from being lowered, only allowing
+		// probability to fall and never to rise.
+		if err := carrier.updateThreshold(threshold); err != nil {
+			if errors.Is(err, sampling.ErrInconsistentSampling) {
+				// This is working-as-intended.  You can't lower
+				// the threshold, it's illogical.
+				logger.Debug(description, zap.Error(err))
```

Comment:
same comment here about the metric vs. log

Reply:
To summarize: I think the existing metric's "policy=missing_randomness" is a useful-enough signal. Personally, I want standard pipeline metrics so that every component doesn't have to invent a bespoke metric definition.